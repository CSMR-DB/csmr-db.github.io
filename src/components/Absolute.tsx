import styled, {
  StyledComponent,
  css,
  FlattenSimpleInterpolation,
} from 'styled-components'
import { ReactNode } from 'react'

export interface IAbsoluteProps {
  children: ReactNode
  $top?: string
  $left?: string
  $bottom?: string
  $right?: string
  $width?: string
  $height?: string
}

export const Absolute: StyledComponent<
  'div',
  any,
  IAbsoluteProps,
  never
> = styled.div`
  position: absolute;

  ${({
    $width: width = 'auto',
    $height: height = 'auto',
    $top: top = '0',
    $left: left = '0',
    $bottom: bottom = '0',
    $right: right = '0',
  }: IAbsoluteProps): FlattenSimpleInterpolation => css`
    width: ${width};
    min-height: ${height};
    top: ${top};
    left: ${left};
    bottom: ${bottom};
    right: ${right};
  `}
`
