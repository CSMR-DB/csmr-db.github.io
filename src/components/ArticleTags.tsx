import React from 'react'
import { StyledLink } from './StyledLink'
import styled, { StyledComponent } from 'styled-components'

const StyledP: StyledComponent<'p', any, {}, never> = styled.p`
  text-align: left !important;
`

export const ArticleTags: ({ tags }: { tags?: string[] }) => JSX.Element = ({
  tags,
}: {
  tags?: string[]
}): JSX.Element => (
  <StyledP>
    Tags:{' '}
    {tags &&
      tags.map(
        (tag: string, i: number): JSX.Element => (
          <span key={i}>
            <StyledLink
              to={`/tag/${tag.toLocaleLowerCase().replace(/\s/gi, '_')}`}
            >
              {tag}
            </StyledLink>
            {i + 1 < tags.length ? ', ' : ''}
          </span>
        )
      )}
  </StyledP>
)
