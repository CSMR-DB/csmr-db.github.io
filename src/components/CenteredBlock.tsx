import styled, { StyledComponent } from 'styled-components'
import { WidthProperty } from 'csstype'

interface ICenteredBlockProps {
  maxWidth?: WidthProperty<1>
}

export const CenteredBlock: StyledComponent<
  'div',
  any,
  ICenteredBlockProps,
  never
> = styled.div`
  max-width: ${({
    maxWidth = '48rem',
  }: ICenteredBlockProps): WidthProperty<1> => maxWidth};
  margin: 0 auto;
`
