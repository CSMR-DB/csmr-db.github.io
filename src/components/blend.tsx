import React, { ReactNode } from 'react'
import styled, { StyledComponent } from 'styled-components'
import { MixBlendModeProperty, IsolationProperty } from 'csstype'

interface IBlend {
  children: ReactNode
  mode: MixBlendModeProperty
  hoverMode?: MixBlendModeProperty
  isolation?: boolean
}

const StyledDiv: StyledComponent<'div', any, IBlend, never> = styled.div`
  position: relative;
  isolation: ${(props: IBlend): IsolationProperty =>
    props.isolation ? 'isolate' : 'auto'};
  overflow: hidden; /** to make sure the containing element has no children overlapping parent's sibling elements */

  & > :first-child {
    position: relative !important;
  }

  :hover > :first-child {
    transform: scale(1.1);
  }

  & > * {
    mix-blend-mode: ${(props: IBlend): MixBlendModeProperty => props.mode};
    position: absolute !important; /** to work with gatsby-image, adding !important is a necessary override */
    width: 100%;
    height: 100%;
    top: 0;
    transition: all 0.25s ease-in-out;

    :hover {
      mix-blend-mode: ${(props: IBlend): MixBlendModeProperty =>
        props.hoverMode || props.mode};
    }
  }
`

export const Blend: ({
  children,
  mode,
  hoverMode,
  isolation,
}: IBlend) => JSX.Element = ({
  children,
  mode,
  hoverMode,
  isolation = false,
}: IBlend): JSX.Element => (
  <StyledDiv mode={mode} isolation={isolation} hoverMode={hoverMode}>
    {children}
  </StyledDiv>
)
