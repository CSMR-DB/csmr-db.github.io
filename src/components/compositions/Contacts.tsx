import React from 'react'

import { Contact } from '../../data/objects/contactsProvider'

import { SocialButton } from './SocialButton'

export interface IContactsProps {
  contacts: Contact[]
}

export function Contacts({ contacts }: IContactsProps): JSX.Element {
  return (
    <>
      {contacts.map(
        (
          { background, iconPath, href, text }: Contact,
          key: number
        ): JSX.Element => (
          <SocialButton
            $background={background}
            iconPath={iconPath}
            href={href}
            key={key}
            $_testid={`SocialButton${text}`}
          >
            {text}
          </SocialButton>
        )
      )}
    </>
  )
}
