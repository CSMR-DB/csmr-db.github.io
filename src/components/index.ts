export { Absolute } from './Absolute'
export { ArticleBody } from './ArticleBody'
export { ArticleTags } from './ArticleTags'
export { Blend } from './Blend'
export {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderFeaturedImage,
  CardHeaderIconWrapper,
  CardHeaderText,
} from './Card'
export { CardHeaderImageOrVideo } from './CardHeaderImageOrVideo'
export { CenteredBlock } from './CenteredBlock'
export { ClippedBtn } from './ClippedBtn'
export { ContentSeparator } from './ContentSeparator'
export { CTAButton } from './CTAButton'
export { DateFormatted } from './DateFormatted'
export { DesktopNav } from './DesktopNav'
export { DynamicIcon } from './DynamicIcon'
export { DynamicImage } from './DynamicImage'
export { ExperienceCard } from './ExperienceCard'
export { Excerpt } from './Excerpt'
export { Filter } from './Filter'
export { Fixed } from './Fixed'
export { Flex } from './Flex'
export { Footer } from './Footer'
export { Grid } from './Grid'
export { Header } from './Header'
export { Heart } from './Heart'
export { Image } from './Image'
export { ImageContainerCircle } from './ImageContainerCircle'
export { ImageDot } from './ImageDot'
export { Layout } from './Layout'
export { Logo } from './Logo'
export { MenuTile } from './MenuTile'
export { ProjectCard } from './ProjectCard'
export { ReadingTime } from './ReadingTime'
export { Relative } from './Relative'
export {
  ScrollIconWrapper,
  AnimatedScroller,
  ScrollText,
} from './ScrollIndicator'
export { SEO } from './SEO'
export { SkillCard } from './SkillCard'
export { SocialButton } from './SocialButton'
export { StyledIFrame } from './StyledIFrame'
export { StyledLink, StyledA } from './StyledLink'
export { WASDNav } from './WASDNav'
