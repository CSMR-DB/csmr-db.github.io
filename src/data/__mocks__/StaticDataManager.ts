import { ExperienceData } from './../graphql/experienceDataProvider'
import { DynamicImages } from './../graphql/dynamicImagesProvider'
import { DynamicIcons } from './../graphql/dynamicIconsProvider'
import { ProjectsData } from '../graphql/projectsDataProvider'
import { SkillsetData } from '../graphql/skillsetDataProvider'
import { GameWallpapers } from '../graphql/gameWallpapersProvider'
import { SeriesPosters } from '../graphql/seriesPostersProvider'
import {
  SiteMetadata,
  SkillsetAllMarkdownRemark,
  ImageSharpAllFiles,
  ProjectsAllMarkdownRemark,
  ProjectMarkdownEdge,
  SkillsetMarkdownEdge,
  ExperienceAllMarkdownRemark,
} from '../../types/graphql.types'
import { PhotographySquares } from '../graphql/photographySquaresProvider'
import { PhotographyData } from '../graphql/photographyDataProvider'
import { Tool } from '../objects/toolsProvider'
import { AStaticDataManager } from '../StaticDataManager'
import { RouteObject } from '../objects/routesProvider'
import { Course } from '../objects/coursesProvider'
import { Contact } from '../objects/contactsProvider'
import { DefaultTheme } from 'styled-components'
import { mockSVGs } from './tools.mocks'

const mockImageCollection: ImageSharpAllFiles = {
  edges: [
    {
      node: {
        name: 'Photo',
        relativePath: '/photo.jpg',
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: '',
            src: '/photo.jpg',
            srcSet: '/photo.jpg',
          },
        },
      },
    },
  ],
}

const mockProgrammingSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'React',
          excerpt: 'I love JSX',
          favorite: true,
          icon: '',
          skillColor: 'red',
          time: 1337,
        },
        generatedRoute: '/skillset/react',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Jest',
          excerpt: 'It Jest needs to be done',
          favorite: true,
          icon: '',
          skillColor: 'blue',
          time: 1336,
        },
        generatedRoute: '/skillset/jest',
      },
    },
  ],
}

const mockProgrammingProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Portfolio',
          excerpt:
            'Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme',
          favorite: true,
          tags: [
            'Gatsby',
            'TypeScript',
            'React',
            'GraphQL',
            'GSAP',
            'Styled-Components',
            'Jest',
            'Enzyme',
          ],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Loopy',
          excerpt: 'I made a fantastic hula-loop once',
          favorite: true,
          tags: ['Hula', 'Loop', 'Perfection'],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
  ],
}

const mockGraphicDesignProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Circle',
          excerpt: 'I drew a fantastic circle once',
          favorite: true,
          tags: ['Pencil', 'Shapes', 'Drawing'],
        },
        generatedRoute: '/projects/circle',
      },
    },
  ],
}

const mockGraphicDesignSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Affinity Designer',
          excerpt: 'It has no proper geometry operations',
          favorite: false,
          icon: '',
          skillColor: 'green',
          time: 420,
        },
        generatedRoute: '/skillset/affinity_designer',
      },
    },
  ],
}

const mockExperienceMdx: ExperienceAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          title: 'React',
          backgroundColor: 'blue',
          type: 'Reactor',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'JSX is just fantastic',
        },
        generatedRoute: '',
      },
    },
    {
      node: {
        frontmatter: {
          title: 'Jest',
          backgroundColor: 'red',
          type: 'Jester',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'Jest a test',
        },
        generatedRoute: '',
      },
    },
  ],
}

export class StaticDataManager implements AStaticDataManager {
  static get theme(): DefaultTheme {
    return {
      palette: {
        dark: {
          normal: '#000',
          hover: '#222',
          active: '#444',
          lens: '#666',
        },
        light: {
          normal: '#fff',
          hover: '#ddd',
          active: '#bbb',
          lens: '#999',
        },
        first: { normal: '#f00', hover: '#d00', active: '#b00', lens: '#900' },
        second: {
          normal: '#0f0',
          hover: '#0d0',
          active: '#0b0',
          lens: '#090',
        },
        third: { normal: '#00f', hover: '#00d', active: '#00b', lens: '#009' },
        filter: {
          normal: 'saturation(4)',
          hover: 'saturation(4)',
          active: 'saturation(4)',
        },
      },
      breakpoints: {
        min: {
          smartphone: 'only screen and (min-width: 600px)',
          tablet: 'only screen and (min-width: 960px)',
          desktop: 'only screen and (min-width: 1200px)',
          ultrawide: 'only screen and (min-width: 1920px)',
        },
        max: {
          smartphone: 'only screen and (max-width: 600px)',
          tablet: 'only screen and (max-width: 960px)',
          desktop: 'only screen and (max-width: 1200px)',
          ultrawide: 'only screen and (max-width: 1920px)',
        },
      },
      fonts: {
        headings: 'Arial',
        copy: 'serif',
      },
    }
  }

  static get siteMetadata(): SiteMetadata {
    return {
      author: 'Jester',
      description: 'The TestRunner',
      image: '/',
      title: 'Jest',
      url: 'https://www.jest.testrunner',
    }
  }

  static get contacts(): Contact[] {
    return [
      {
        background: 'red',
        href: 'https://mock.test.nowhere',
        iconPath: '',
        text: 'Mock Option',
      },
    ]
  }

  static get courses(): Course[] {
    return [
      {
        category: 'Development',
        description: 'Experimenting with vanilla JS & jQuery',
        name: 'WebProgramming',
      },
    ]
  }

  static get routes(): RouteObject[] {
    return [
      {
        path: '/route',
        title: 'Route',
        boundKeys: ['R', 1],
      },
    ]
  }

  static get tools(): Tool[] {
    return [
      {
        name: 'React',
        svg: mockSVGs().RandomSampleSVG,
        color: 'red',
      },
    ]
  }

  static get photographyData(): PhotographyData {
    return {
      photographyData: mockImageCollection,
    }
  }

  static get photographySquares(): PhotographySquares {
    return {
      photographySquares: mockImageCollection,
    }
  }

  static get seriesPosters(): SeriesPosters {
    return {
      seriesPosters: mockImageCollection,
    }
  }

  static get gameWallpapers(): GameWallpapers {
    return {
      gameWallpapers: mockImageCollection,
    }
  }

  static get skillsetData(): SkillsetData {
    return {
      all: { ...mockProgrammingSkillsetMdx, ...mockGraphicDesignSkillsetMdx },
      top: {
        edges: [
          ...mockProgrammingSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
          ...mockGraphicDesignSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingSkillsetMdx,
      allGraphicDesign: mockGraphicDesignSkillsetMdx,
    }
  }

  static get projectsData(): ProjectsData {
    return {
      all: { ...mockGraphicDesignProjectsMdx, ...mockProgrammingProjectsMdx },
      top: {
        edges: [
          ...mockGraphicDesignProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
          ...mockProgrammingProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingProjectsMdx,
      allGraphicDesign: mockGraphicDesignProjectsMdx,
    }
  }

  static get experienceData(): ExperienceData {
    return {
      experienceData: mockExperienceMdx,
    }
  }

  static get dynamicIcons(): DynamicIcons {
    return {
      dynamicIcons: mockImageCollection,
    }
  }

  static get dynamicImages(): DynamicImages {
    return {
      dynamicImages: mockImageCollection,
    }
  }
}
