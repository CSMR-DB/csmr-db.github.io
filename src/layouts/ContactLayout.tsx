import React from 'react'
import { SEO } from '../components/SEO'
import { Grid } from '../components/Grid'
import { SocialButton } from '../components/SocialButton'
import { CenteredBlock } from '../components/CenteredBlock'

import { BackgroundColorProperty } from 'csstype'

export interface IContact {
  background: BackgroundColorProperty
  iconPath: string
  href: string
  text: string
}

export interface IContactLayoutProps {
  contacts: IContact[]
}

export function ContactLayout({ contacts }: IContactLayoutProps): JSX.Element {
  return (
    <>
      <SEO
        title="Contact"
      // description="Get in touch with me through some social links"
      />
      <CenteredBlock>
        <h1>Get in touch!</h1>
        <p>
          I don't do Twitter and I don't really use Facebook other than checking
          it about twice a year. However, there are other ways to get in touch
          with me. I don't have a fancy form (yet), but that shouldn't be a
          dealbreaker. I check my mail at least once a day, LinkedIn weekly and
          I'm pretty much always available on Discord. You just do whatever feels
          right ðŸ˜‰.
      </p>
        <Grid columns={3} rows={1}>
          {contacts.map(
            ({ background, iconPath, href, text }: IContact, key: number): JSX.Element => (
              <SocialButton
                background={background}
                iconPath={iconPath}
                href={href}
                key={key}
              >
                {text}
              </SocialButton>
            )
          )}
        </Grid>
      </CenteredBlock>
    </>
  )
}