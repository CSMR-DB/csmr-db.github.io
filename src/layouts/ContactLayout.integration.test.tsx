import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'

import { StaticDataManager } from '../data/StaticDataManager'

import { ContactLayout } from './ContactLayout'
import { CenteredBlock } from '../components/CenteredBlock'
import { Grid } from '../components/Grid'
import { Contacts } from '../components/compositions/Contacts'
import { Layout } from '../components/Layout'

describe('ContactLayout', (): void => {
  it('should match snapshot', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <ContactLayout contacts={StaticDataManager.contacts} />
      </ThemeProvider>
    )

    expect(comp).toMatchSnapshot()
  })

  it('should render internally', (): void => {
    const comp: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>> = mount(
      <ThemeProvider theme={StaticDataManager.theme}>
        <ContactLayout contacts={StaticDataManager.contacts} />
      </ThemeProvider>
    )

    expect(comp.find(Layout).props().isLandingPage).not.toEqual(true)
    expect(comp.find(CenteredBlock).at(0).find('h1').text()).toEqual(
      'Get in touch!'
    )
    expect(comp.find(CenteredBlock).at(0).find(Grid).length).toEqual(1)
    expect(
      comp.find(CenteredBlock).at(0).find(Grid).find(Contacts).length
    ).toEqual(1)
  })
})
