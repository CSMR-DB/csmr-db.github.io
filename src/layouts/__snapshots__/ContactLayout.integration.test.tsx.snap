// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactLayout should match snapshot 1`] = `
.c1 {
  margin: 0 auto;
  padding: 1rem;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f00;
}

.c1 a:hover {
  color: #d00;
}

.c0 {
  min-height: 100vh;
  background: #fff;
}

.c3 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: auto;
  gap: calc(1rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: repeat(1,1fr);
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

.c2 {
  margin: 0 auto;
  max-width: 48rem;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: #f00 !important;
}

.c4:hover {
  color: #d00;
}

.c4.active {
  color: #b00;
}

.c5 {
  position: relative;
}

.c7 {
  position: absolute;
  width: auto;
  min-height: auto;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.c8 {
  height: 100%;
  border-radius: 0.25rem;
  -webkit-transition: color 0.1s ease-in 0.1s;
  transition: color 0.1s ease-in 0.1s;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  font-size: 1rem;
  color: white;
  background: red;
}

.c9 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 100%;
  width: 100%;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c6 {
  margin: 0 auto;
  max-width: 4rem;
}

@media only screen and (min-width:600px) {
  .c1 {
    padding: 4rem 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c3 {
    grid-template-columns: repeat(3,1fr);
    gap: 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c8 {
    -webkit-clip-path: circle(16em at 50% 50%);
    clip-path: circle(16em at 50% 50%);
  }
}

@media only screen and (min-width:960px) {
  .c8 {
    -webkit-clip-path: circle(1em at 50% 50%);
    clip-path: circle(1em at 50% 50%);
    font-size: 2rem;
  }

  .c8:hover {
    -webkit-clip-path: circle(8em at 50% 50%);
    clip-path: circle(8em at 50% 50%);
  }
}

<div
  className="c0"
>
  <main
    className="c1"
  >
    <div
      className="c2"
    >
      <h1>
        Get in touch!
      </h1>
      <p>
        I don't do Twitter and I don't really use Facebook other than checking it about twice a year. However, there are other ways to get in touch with me. I don't have a fancy form (yet), but that shouldn't be a dealbreaker. I check my mail at least once a day, LinkedIn weekly and I'm pretty much always available on Discord. You just do whatever feels right ðŸ˜‰.
      </p>
      <div
        className="c3"
      >
        <a
          className="c4"
          href="https://mock.test.nowhere"
          target="blank"
        >
          <div
            className="c5"
          >
            <div
              className="c6"
            />
            <div
              className="c7"
            >
              <div
                className="c8"
              >
                <div
                  className="c9"
                >
                  <span>
                    Mock Option
                  </span>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </main>
</div>
`;
